### ISSUE TITLE
stability issue with equal order in fractional step method

### ISSUE THREAD

**Issue by yuxianglin** [mikaem/Oasis#26]
Hi,

Just curious in the equi-order of velocity and pressure setting in problem/Channel, taylorgreen,  will the LBB condition be satisfied? Does the Oasis use any specific stabilization trick?

Shawn

**Comment by mikaem** [comment]
Hi,

The LBB is only relevant for coupled solvers, there's no need for stabilization tricks for segregated solvers.

**Comment by yuxianglin** [comment]
Well, this is not very true, some papers claim equi-order scheme in projection method is not very stable for small time step, see AN OVERVIEW OF PROJECTION METHODS FOR IN-COMPRESSIBLE
FLOWS.

**Comment by mikaem** [comment]
I am well aware of that paper. And for steady state problems there may be issues I have still not encountered. But for transient problems, like the channel and the Taylor Green, there are no instabilities observed in the solution. See for example the P1P1 results for TG in the [Oasis paper](https://www.sciencedirect.com/science/article/pii/S0010465514003786).  Note that we take care of the pressure nullspace in Oasis by [modifying the Krylov solver directly](https://github.com/mikaem/Oasis/blob/3de552ca7271b0f91bed51322bab4fce06e63a99/oasis/solvers/NSfracStep/IPCS_ABCN.py#L204), and with this modification the spurious pressure modes are removed.

**Comment by yuxianglin** [comment]
remove pressure nullspace you mean to avoid p and p+c to be the solution when solve a pure neumman problem in pressure poission equation?

**Comment by yuxianglin** [comment]
I tried cavity driven flow with velocity degree=1 in Oasis, but I got spurious pressure and velocity

**Comment by mikaem** [comment]
Yes, that's a good example of a steady state problem that should not use P1P1. I remember now that I have seen those oscillations as well. For steady state problems I always recommend using a coupled solver.

**Comment by yuxianglin** [comment]
For transient problem too, like the flow around cylinder case, the pressure do not show very good result, but velocity does perform well

**Comment by mikaem** [comment]
I think you can get rid of these wiggles using a different velocity update method, like [lumping or weighted gradient](https://github.com/mikaem/Oasis/blob/3de552ca7271b0f91bed51322bab4fce06e63a99/oasis/problems/NSfracStep/__init__.py#L61). It has to do with how the velocity divergence is projected. I haven't looked at this for quite some time, though, so not sure the non-default methods will work out of the box. Weighted gradient is using fenistools and I'm not sure it has been updated to latest version of fenics.

**Comment by mikaem** [comment]
I also think you should try a longer timestep. That will probably make the wiggles disappear.

**Comment by yuxianglin** [comment]
Well, I tried lump mass or weigted gradient but the wiggle still exist for P1P1 element though. Since I wanted to have more time resolve of the fluid, large dt isn't an option.

**Comment by mikaem** [comment]
But large dt gets rid of the wiggles, right?

**Comment by yuxianglin** [comment]
no, increase dt cann't completely remove the wiggle. Only slightly better

**Comment by mikaem** [comment]
Which solver is this?

**Comment by yuxianglin** [comment]
the cylinder problem with Von Karman vortex street, the solver is NSfracstep with P1P1
